commit 83881990b56489e2310f8edf33313b5247b5f1e9
Author: Isuru Fernando <isuruf@gmail.com>
Date:   Sun Sep 23 13:45:52 2018 -0500

    S_IRWXU is not defined on msvc

diff --git a/tests/hwloc/hwloc_backends.c b/tests/hwloc/hwloc_backends.c
index f969c45c..6e584cd5 100644
--- a/tests/hwloc/hwloc_backends.c
+++ b/tests/hwloc/hwloc_backends.c
@@ -22,7 +22,11 @@
 static inline int mkstemp(char *name)
 {
   mktemp(name);
+#ifdef _MSC_VER
+  return open(name, O_RDWR|O_CREAT, 00700);
+#else
   return open(name, O_RDWR|O_CREAT, S_IRWXU);
+#endif
 }
 #endif
 
