diff --git a/doc/examples/cpuset+bitmap+cpubind.c b/doc/examples/cpuset+bitmap+cpubind.c
index 9e0ae64..a025442 100644
--- a/doc/examples/cpuset+bitmap+cpubind.c
+++ b/doc/examples/cpuset+bitmap+cpubind.c
@@ -11,7 +11,9 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#endif
 
 int main(void)
 {
diff --git a/doc/examples/nodeset+membind+policy.c b/doc/examples/nodeset+membind+policy.c
index 21fffb6..5123629 100644
--- a/doc/examples/nodeset+membind+policy.c
+++ b/doc/examples/nodeset+membind+policy.c
@@ -11,7 +11,9 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#endif
 #include <assert.h>
 
 int main(void)
diff --git a/doc/examples/sharedcaches.c b/doc/examples/sharedcaches.c
index 91b6af4..f821f37 100644
--- a/doc/examples/sharedcaches.c
+++ b/doc/examples/sharedcaches.c
@@ -9,7 +9,11 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#else
+#define pid_t int
+#endif
 
 int main(int argc, char *argv[])
 {
